src/contracts/Output/Generator.php

```diff
@@ -4,65 +4,90 @@
  * @copyright Copyright (C) Ibexa AS. All rights reserved.
  * @license For full copyright and license information view LICENSE file distributed with this source code.
  */
+declare(strict_types=1);
 
 namespace Ibexa\Contracts\Rest\Output;
 
+use Ibexa\Rest\Output\Generator\Data;
+use Ibexa\Rest\Output\Generator\Json;
+use Ibexa\Rest\Output\Generator\Xml;
+
 /**
  * Output generator.
  */
 abstract class Generator
 {
+    /**
+     * Keeps track if the document is still empty.
+     */
+    protected bool $isEmpty = true;
+
     /**
      * Generator creation stack.
      *
      * Use to check if it is OK to start / close the requested element in the
      * current state.
-     *
-     * @var array
      */
-    protected $stack = [];
+    protected array $stack = [];
 
     /**
      * If set to true, output will be formatted and indented.
-     *
-     * @var bool
      */
-    protected $formatOutput = false;
+    protected bool $formatOutput = false;
+
+    /**
+     * Enables developer to modify REST response media type prefix.
+     */
+    protected string $vendor;
+
+    /**
+     * Generator for field type hash values.
+     */
+    protected Json\FieldTypeHashGenerator|Xml\FieldTypeHashGenerator $fieldTypeHashGenerator;
```

Comment:
```suggestion
    protected \Ibexa\Rest\Output\Generator\AbstractFieldTypeHashGenerator $fieldTypeHashGenerator;
```
Alternatively, introduce an interface for it.